### ISSUE TITLE
[ts] TypeScript type definitions hide null values

### ISSUE THREAD

**Issue by mvirkkunen** [EsotericSoftware/spine-runtimes#2058]
It seems you have not enabled strict mode for TypeScript, which results in really weird type signatures, such as:

```
	/** Sets the skin (...)
	 * @param newSkin May be null. */
	setSkin (newSkin: Skin) {
```

```
	/** The skeleton's current skin. May be null. */
	skin: Skin = null;
```

The documentation comments state that values "may be null", while the `Skin` type is definitely not allowed to be `null`. The same problem is also present in many other places.

If strict null checks are enabled (as they should be) in a project using this library, it results in many potential hazards that could be prevented at compile time when properties lie about their type and state they can't be null, even though they can be. Additionally calling e.g. `setSkin` will a `null` value which is allowed according to documentation requires a really unsafe-looking cast.

**Comment by badlogic** [comment]
We have indeed not enabled strict mode in TypeScript for historical reasons. You are right that the API should behave properly when consuming code has strict null checks enabled. This will be fixed for the 4.1 release.

**Comment by jonlepage** [comment]
a temp fix for me, i monkey-patch a deep partial 😆.
Will be a issue for enumerable instances but we rem lints error.
```ts
type DeepPartial<T> = {
    [P in keyof T]?: DeepPartial<T[P]>;
};
```
i also wait for the fix thanks. :)

**Comment by badlogic** [comment]
I should have this fixed in the 4.1-beta branch later today. 67 errors remaining...

**Comment by badlogic** [comment]
This is now implemented in 4.1-beta.
